components/app-proxy/vm.tf

```diff
@@ -6,78 +6,75 @@ module "tags" {
   builtFrom   = var.builtFrom
 }
 
+
 resource "azurerm_resource_group" "this" {
   name     = "${var.product}-${var.env}-rg"
   location = var.location
 
   tags = module.tags.common_tags
 }
 
-# TODO store in a vault
-resource "random_password" "this" {
-  length = 30
+resource "azurerm_subnet" "app_proxy_subnet" {
+  name                 = "app-proxy"
+  address_prefixes     = ["10.99.73.0/25"]
+  resource_group_name  = var.app_proxy_subnet_rg
+  virtual_network_name = var.app_proxy_subnet_name
 }
 
-resource "azurerm_subnet" "this" {
-  name = "app-proxy"
+data "azurerm_key_vault" "reform_mgmt_kv" {
+  name                = "ReformMgmtKeyVault"
+  resource_group_name = "reformMgmtCoreRG"
+}
 
-  address_prefixes     = ["10.99.73.0/25"]
-  resource_group_name  = "mgmt-vpn-2-mgmt"
-  virtual_network_name = "mgmt-vpn-2-vnet"
+data "azurerm_key_vault_secret" "vm_admin_password" {
```

Comment:
this shouldn't be a data source, terraform should generate the password and then store it in key vault